### PR TITLE
update 2.0 public api in optimizer

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

Others
### PR changes

APIs
### Describe

update 2.0 public api in optimizer

背景：
对于用户调用api的层级，由于之前没有明确的设计，api开放层级比较随意。影响用户使用中的逻辑和正规感受。
目的：
1）重新整理api开放层级，提升用户使用感受
2）API保留自有特色，同时更好地符合行业规范，降低用户学习成本
3）提高框架api逻辑性，便于后续维护和优化
主要实现方法：
1）__all__内容清晰化，内容逐个列出，一行一个。不再使用 __all__ 的向上传递机制，仅在公开API层级添加 __all__，其余的 __all__ 列表统一置空
2）不再使用 #DEFINE_ALIAS
3）导入内容清晰化，原则上不再使用 import *， 按需引入需要的API， 每行一个
4）原则上不再使用 from . import module， from .module import submodule时会引入module，避免重复引用
5）添加新的Public API时，需向上寻找到最近的节点添加 import 并加入 __all__ 列表
6）Public API列表由多个公开API层级的 __all__ 列表共同组成，文档抽取以此为准，这些指定层级是：
- paddle
- paddle.amp
- paddle.nn
- paddle.nn.functional
- paddle.nn.initializer
- paddle.nn.utils
- paddle.static
- paddle.static.nn
- paddle.io
- paddle.jit
- paddle.metric
- paddle.distribution
- paddle.optimizer
- paddle.optimizer.lr
- paddle.regularizer
- paddle.text
- paddle.vision
- paddle.callbacks
- paddle.distributed
- paddle.distributed.fleet
- paddle.distributed.fleet.base.distributed_strategy
- paddle.distributed.fleet.data_generator
- paddle.distributed.fleet.utils
- paddle.distributed.parallel
- paddle.distributed.utils
- paddle.sysconfig
- paddle.utils
- paddle.utils.download
- paddle.utils.profiler
修改示例展示：
原来写法：
image
新的写法：
image

对开发者使用API的影响：
在指定层级推荐方式：（不受影响）
  from paddle.nn import CTCLoss
  from paddle.nn import Conv2D
非指定层级支持但不推荐：（不受影响）
  from paddle.nn.layer.loss import CTCLoss
  from paddle.nn.layer.conv import Conv2D
对于import *使用方式不推荐。（非公开层级受影响）
  不再支持：（非公开层级）
  from paddle.nn.layer import *
  from paddle.nn.layer.conv import *
  from paddle.nn.initializer.norm import *
  from paddle.tensor.math import *
  from paddle.optimizer.adam import *
  可以支持：（公开层级）
  from paddle import *
  from paddle.nn import *
  from paddle.utils import *
  from paddle.optimizer import *

本pr涉及到的api改动：
1）paddle.optimizer下__all__列表中移除lr。
使用方式的差别：
修改前可以用：
from paddle.optimizer import *
lr.SGD()
修改后只支持：
from paddle.optimizer import lr
lr.SGD()

本pr其它修改对api的调用无明显影响

### CODE REVIEW

File: python/paddle/optimizer/__init__.py [hunk 1]
```diff
@@ -12,19 +12,32 @@
...
+           'Optimizer', 
+           'Adagrad', 
+           'Adam', 
+           'AdamW', 
+           'Adamax', 
+           'RMSProp', 
+           'Adadelta', 
+           'SGD', 
+           'Momentum', 
+           'lr',
```
Comment (willthefrog) [inline python/paddle/optimizer/__init__.py]: is this in the list? anyway, should remove it

**Review by willthefrog — CHANGES_REQUESTED** [review]

File: python/paddle/optimizer/__init__.py (hunk 1 above)
Comment (zhiboniu) [reply python/paddle/optimizer/__init__.py]: done

File: python/paddle/optimizer/__init__.py [hunk 2]
```diff
@@ -12,19 +12,30 @@
...
+           'AdamW',
+           'Adamax',
+           'RMSProp',
+           'Adadelta',
+           'SGD',
+           'Momentum',
+           'append_gradient_clip_ops',
+           'find_distributed_lookup_table',
+           'error_clip_callback',
+           'append_regularization_ops'
```
Comment (jzhang533) [inline python/paddle/optimizer/__init__.py]: 列表里的最后这4个API，原来是没有暴露的吧？
这次加上的原因是？

Comment (zhiboniu) [reply python/paddle/optimizer/__init__.py]: 按2.0api列表整理时家上来的，看起来是比较旧的api，已移除

**Review by willthefrog — DISMISSED** [review]

File: python/paddle/optimizer/__init__.py [hunk 3]
```diff
@@ -12,19 +12,26 @@
...
+__all__ = [     #noqa
+           'Optimizer',
+           'Adagrad',
+           'Adam',
+           'AdamW',
+           'Adamax',
+           'RMSProp',
+           'Adadelta',
+           'SGD',
+           'Momentum'
```
Comment (jzhang533) [inline python/paddle/optimizer/__init__.py]: 是不是落下了`Lamb`?

Comment (zhiboniu) [reply python/paddle/optimizer/__init__.py]: done, 已添加

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by iclementine — APPROVED** [review]
LGTM
